### PR TITLE
Look for matching containers by containerId only when our CloudCache has a daemon running in it

### PR DESCRIPTION
**Pull request by nick4598**

### CODE REVIEW

**Comment by nick4598** [comment]
@mergifyio backport release/4.x

File: iModelCore/BeSQLite/CloudSqlite.cpp [hunk 1]
```diff
@@ -227,8 +227,11 @@ CloudResult CloudCache::CallSqliteFn(std::function<int(Utf8P*)> fn, Utf8CP funcN
 }
 
 CloudContainer* CloudCache::FindMatching(Utf8CP baseUri, Utf8CP containerName) {
+    auto isDaemon = this->IsDaemon();
     for (auto entry : m_containers) {
-        if (entry->m_containerId.Equals(containerName) && entry->m_baseUri.Equals(baseUri))
+        if (isDaemon && entry->m_containerId.Equals(containerName)) 
+            return entry;
+        else if (!isDaemon && entry->m_containerId.Equals(containerName) && entry->m_baseUri.Equals(baseUri))
```
Comment (kabentley) [inline iModelCore/BeSQLite/CloudSqlite.cpp:234]: Probably worth a comment explaining that for daemon we presume that all containers are from the same storage account for a single process.

remove `else`.

Comment (nick4598) [reply iModelCore/BeSQLite/CloudSqlite.cpp]: Changed it

File: iModelCore/BeSQLite/CloudSqlite.cpp [hunk 2]
```diff
@@ -227,8 +227,12 @@ CloudResult CloudCache::CallSqliteFn(std::function<int(Utf8P*)> fn, Utf8CP funcN
 }
 
 CloudContainer* CloudCache::FindMatching(Utf8CP baseUri, Utf8CP containerName) {
+    auto isDaemon = this->IsDaemon();
     for (auto entry : m_containers) {
-        if (entry->m_containerId.Equals(containerName) && entry->m_baseUri.Equals(baseUri))
+        // For the daemon we presume that all containers are from the same storage account for a single process. No need to check baseUri in this case.
```
Comment (kabentley) [inline iModelCore/BeSQLite/CloudSqlite.cpp:232]:
```c
if ((entry->m_containerId.Equals(containerName) && (isDaemon || entry->m_baseUri.Equals(baseUri))
    return entry;
```

Comment (nick4598) [reply iModelCore/BeSQLite/CloudSqlite.cpp:232]: changed it

**Review by kabentley — APPROVED** [review]

**Comment by nick4598** [comment]
@mergifyio backport release/3.x

**Pull request merged**